casper4/papers/casper-basics/casper_basics.tex

```diff
@@ -137,19 +137,33 @@ \section{The Protocol}
 
  A \textit{checkpoint for epoch $n$} is a block with number $n * 100 - 1$; in a smoothly running blockchain there will usually be only one checkpoint per epoch, but due to natural network latency or deliberate attacks there may be multiple competing checkpoints during some epochs. The \textit{predecessor} of a checkpoint is the checkpoint in the prior epoch, and an \textit{lineage} of a checkpoint is the set of all preceding checkpoints, recursively, to the Genesis block \Genesisblock.
  
-\TODO{Probably define some mathematical notation here for Block $B$, Checkpoint $C$, Predecessor, Successor, Lineage, etc.}
 
-We define the \textit{lineage hash} of a checkpoint as follows:
+Our atomic element of construction is $B$, a \emph{blockhash}. We order a sequence of blockhashes into a list called a \emph{blockchain} $\mathbf{B}$.  Within blockchain $\mathbf{B}$ there is a subset called \emph{checkpoints},
```

Comment:
> We order a sequence of blockhashes into a list called a \emph{blockchain} $\mathbf{B}$.
= >
> We order a sequence of blockhashes into a list called a \emph{blockchain}, $(\mathbf{B})*$, where * indicates any natural number of repetitions.

Reply:
Would be good to also mention that the hash function is the Keccak 256, probably the original implementation, e.g. as in the Yellow Paper: `the Keccak-256 hash function (as per the winning entry to the SHA-3 contest by \cite{Keccak}, rather than later releases)`.

```
@misc{Keccak,
	url = "https://keccak.team/keccak.html",
	note = "Accessed 6 October 2017. Unable to be archived by the Wayback Machine.",
	author = "Guido Bertoni
and Joan Daemen
and Michaël Peeters
and Gilles Van Assche
and Ronny Van Keer",
	title = {{KECCAK}},
	year = "2017",
}
```